spring-batch-integration/src/main/java/org/springframework/batch/integration/config/annotation/EnableBatchIntegration.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.batch.integration.config.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.batch.integration.chunk.RemoteChunkingWorkerBuilder;
+import org.springframework.context.annotation.Import;
+
+/**
+ * Enable Spring Batch Integration features and provide a base configuration for
+ * setting up remote chunking infrastructure beans.
+ *
+ * By adding this annotation on a {@link org.springframework.context.annotation.Configuration}
+ * class, it will be possible to autowire the following beans:
+ *
+ * <ul>
+ *     <li>{@link org.springframework.batch.integration.chunk.RemoteChunkingMasterStepBuilderFactory}:
+ *     used to create a master step by automatically setting up the job repository,
+ *      transaction manager and application context.</li>
+ *     <li>{@link RemoteChunkingWorkerBuilder}:
+ *     used to create the integration flow on the worker side.</li>
+ * </ul>
+ *
+ * For example:
+ *
+ * <pre class="code">
+ * &#064;Configuration
+ * &#064;EnableBatchIntegration
+ * public class AppConfig {
+ *
+ * 	&#064;Autowired
+ * 	private RemoteChunkingMasterStepBuilderFactory masterBuilderFactory;
+ *
+ * 	&#064;Autowired
+ * 	private RemoteChunkingWorkerBuilder workerBuilder;
```

Comment:
Well, indeed, looking to this sample, it really would feel better to have `RemoteChunkingMasterStepBuilder` and `RemoteChunkingWorkerBuilder` - transparent and intuitive. 
Otherwise if I would inject `RemoteChunkingMasterStepBuilderFactory`, I would expect similar for the worker.

Reply:
You are right, there is a factory on the master side but not on the worker side.. Because the worker side does not know anything about SB artifacts (step, job repository, etc) except a dumb item processor and writer.